src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -331,6 +338,196 @@ def db_update(
     return instance
 
 
+#####
+#           sql server audit-policy & threat-policy
+#####
+
+
+# Finds a storage account's resource group by querying ARM resource cache.
+# Why do we have to do this: so we know the resource group in order to later query the storage API
+# to determine the account's keys and endpoint. Why isn't this just a command line parameter:
+# because if it was a command line parameter then the customer would need to specify storage
+# resource group just to update some unrelated property, which is annoying and makes no sense to
+# the customer.
+def _find_storage_account(name):
+    resource_type = 'Microsoft.Storage/storageAccounts'
+
+    client = get_mgmt_service_client(ResourceManagementClient)
+    resources = list(client.resources.list(
+        filter="name eq '{}' and resourceType eq '{}'"
+        .format(name, resource_type)))
+
+    if len(resources) == 0:
+        raise CLIError('No resource with name {} and type {} was found.'
+                       .format(name, resource_type))
+
+    if len(resources) > 1:
+        raise CLIError('Multiple resources with name {} and type {} were found.'
+                       .format(name, resource_type))
+
+    # Split the uri and return just the resource group
+    return resources[0].id.split('/')[4]
+
+
+# Determines storage account name from endpoint url string.
+# e.g. 'https://mystorage.blob.core.windows.net' -> 'mystorage'
+def _get_storage_account_name(storage_endpoint):
+    return urlparse(storage_endpoint).netloc.split('.')[0]
+
+
+# Gets storage account key by querying storage ARM API.
+def _get_storage_endpoint(
+        storage_account,
+        resource_group_name):
+
+    # Get storage account
+    client = get_mgmt_service_client(StorageManagementClient)
+    account = client.storage_accounts.get_properties(
+        resource_group_name=resource_group_name,
+        account_name=storage_account)
+
+    # Get endpoint
+    return account.primary_endpoints.blob  # pylint: disable=no-member
```

Comment:
Ok. I'm not sure how to create such a storage account, so I'm taking your word for it ;) I tested by replacing `primary_endpoints.blob` with `primary_endpoints.potato` and then handling that exception.